### DISCUSSION TITLE
How to create child element for edges

### DISCUSSION THREAD

**Discussion by sathya-1994** [eclipse-glsp/glsp#748] [Q&A]
I am using vscode version of GLSP,

I want to create circle icon similar to routing points of the edges(create a new circle icon with edge as parent).
From GGraph model, I am creating the graph object at server side. I tried to add org.eclipse.glsp.graph.impl.GNodeImpl elemnt as child of org.eclipse.glsp.graph.impl.GEdgeImpl, but child element(GNodeImpl) was not visible in the UI.

Screesnhot:
![image](https://user-images.githubusercontent.com/59145404/191725290-1f7c7c42-c902-4fe0-b550-84d769dd16ba.png)

**How we can implement my requirement?**

**Comment by martin-fleck-at** [comment]
@sathya-1994 Could you please elaborate on the problem a bit more. When you say 'it was not visible in the UI', was the newly created circle element was rendered as element in the SVG? What type did you use for your child? By default a node is using a 'node' type and should be rendered as a (rounded corner) node so you should at least see a 'path' element in the SVG.

We perform a similar operation on the client side when the user clicks on an edge that can be re-connected. In this case we add a reconnect handle at the start and beginning of the edge so the user can drag to reconnect the edge. You can find the code response in the [reconnect model](https://github.com/eclipse-glsp/glsp-client/blob/6bee8ebad7771ab470b4e7299fc8f2c0856b54b5/packages/client/src/features/reconnect/model.ts#L41) and the [command that adds the handles to the edge](https://github.com/eclipse-glsp/glsp-client/blob/b3a810f8c3d8f0fe28982c74079c432e54624b35/packages/client/src/features/tool-feedback/edge-edit-tool-feedback.ts#L98).
